### ISSUE TITLE
gradient propogation

### ISSUE THREAD

**Issue by zhen0z** [sintefmath/JutulDarcy.jl#117]
Hi Olav,

If I get gradient of waveform data with respect to saturation (dL / dsw), can I set dL / dsw as the adjoint source to compute the gradient of waveform data with respect to permeability? 

```
    function objective_function(model, state, Δt, step_i, forces)
        loss = 0.0
        loss += sum( dL / dsw .^ 2 )
        return loss
    end
```
Thanks.

**Comment by zhen0z** [comment]
Yes, I've solved this problem using an automatic differentiation framework. Thanks.
